recipe/meta.yaml

```
@@ -1,44 +1,50 @@
 {% set name = "datasets" %}
-{% set version = "2.12.0" %}
+{% set version = "2.19.1" %}
 
 package:
   name: {{ name|lower }}
   version: {{ version }}
 
 source:
   url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
-  sha256: faf164c18a41bea51df3f369e872f8be5b84c12ea5f6393c3896f56038af1ea3
+  sha256: 0df9ef6c5e9138cdb996a07385220109ff203c204245578b69cca905eb151d3a
+  patches:
+    # removes pyarrow_hotfix as a requirement
+    # hotfix is included in pyarrow
+    - patches/0001_remove_pyarrow_hotfix.patch
 
 build:
-  # s390x is missing pyarrow atm 
-  skip: true  # [py<37 or s390x] 
+  skip: true  # [py<38]
   entry_points:
     - datasets-cli=datasets.commands.datasets_cli:main
   number: 0
   script: {{ PYTHON }} -m pip install . --no-deps --no-build-isolation --ignore-installed -vv
 
 requirements:
+  build:
+    - patch       # [not win]
+    - m2-patch    # [win]
   host:
     - python
     - pip
     - setuptools
     - wheel
   run:
     - python
+    - filelock
     - numpy >=1.17
-    - pyarrow >=8.0.0
-    - dill >=0.3.0,<0.3.7
+    - pyarrow >=12.0.0
+    - dill >=0.3.0,<0.3.9
     - pandas
     - requests >=2.19.0
     - tqdm >=4.62.1
     - python-xxhash
     - multiprocess
     - importlib-metadata  # [py<38]
-    - fsspec >=2021.11.1
+    - fsspec >=2023.1.0,<=2024.3.1
     - aiohttp
-    - huggingface_hub >=0.11.0,<1.0.0
+    - huggingface_hub >=0.21.2
     - packaging
-    - responses <0.19
     - pyyaml >=5.1
```

```suggestion
    - pyyaml >=5.1
  run_constrained:
    - apache-beam >=2.26.0
    - tensorflow-base >=2.6.0
    - jax >=0.3.14
    - jaxlib >=0.3.14
    - soundfile >=0.12.1
    - pillow >=6.2.1
```
Adding run_constrained from https://github.com/huggingface/datasets/blob/2.19.1/setup.py#L273